File: orm-reference-implementations/entity-framework/entity/StudyProgram.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace university.entity
{
    [Table("study_program")]
    public class StudyProgram : NamedElement
    {
        [Column(TypeName = "Varchar(255)")]
        public StudyProgramType? StudyProgramType { get; set; }

        public List<StudentStudyProgram>? Students { get; set; }

        public List<StudentCardStudyProgram>? StudentCards { get; set; }

    }
}

File: orm-reference-implementations/hibernate/src/main/java/entity/StudyProgram.java
package entity;

import jakarta.persistence.Column;
import jakarta.persistence.Convert;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import java.util.List;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class StudyProgram extends NamedElement {

  @Convert(converter = StudyProgramType.Converter.class)
  @Column(name = "study_program_type")
  private StudyProgramType studyProgramType;

  @OneToMany(mappedBy = "studyProgram")
  private List<StudentStudyProgram> students;

  @OneToMany(mappedBy = "studyProgram")
  private List<StudentCardStudyProgram> studentCards;

}
